jupyter_server/terminal/handlers.py

```diff
@@ -18,8 +20,14 @@ def origin_check(self):
         return True
 
     def get(self, *args, **kwargs):
-        if not self.get_current_user():
+        user = self.current_user
+
+        if not user:
             raise web.HTTPError(403)
```

Comment:
Should probably also be 401? Would need to scan for other existing 403s that can be incorrect as well.